### PR TITLE
Add tests for additional uses of @@toPrimitive

### PR DESCRIPTION
**Pull request by jugglinmike**

### CODE REVIEW

**Comment by leobalter** [comment]
oh, no! #666 does not add tests for `with`...

reviewing

File: test/language/expressions/addition/coerce-symbol-to-prim-err.js [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+---*/
+
+var thrower = {};
+var counter = {};
+var log;
+
+Object.defineProperty(thrower, Symbol.toPrimitive, {
+  get: function() {
+    log += 'accessThrower'
+    return function() { throw new Test262Error(); }
```
Comment (leobalter) [inline test/language/expressions/addition/coerce-symbol-to-prim-err.js]: missing semicolon, both of the 2 lines above

File: test/language/expressions/addition/coerce-symbol-to-prim-err.js [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+  }
+});
+Object.defineProperty(counter, Symbol.toPrimitive, {
+  get: function() {
+    log += 'accessCounter';
+    return function() { log += 'callCounter'; }
```
Comment (leobalter) [inline test/language/expressions/addition/coerce-symbol-to-prim-err.js]: another missing semicolon

File: test/built-ins/Date/value-symbol-to-prim-invocation.js [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+var callCount = 0;
+var thisVal, args;
+
+y[Symbol.toPrimitive] = function() {
+  callCount += 1;
+  thisVal = this;
+  args = arguments;
+};
+
+0 == y;
```
Comment (leobalter) [inline test/built-ins/Date/value-symbol-to-prim-invocation.js]: is this test related to Date? Maybe this line should be replaced with `new Date(y);` ?

**Comment by leobalter** [comment]
Just a few comments above. Once they're fixed, they'll be good to land.

File: test/built-ins/Date/value-symbol-to-prim-invocation.js (hunk 3 above)
Comment (jugglinmike) [inline test/built-ins/Date/value-symbol-to-prim-invocation.js]: Definitely. Must have fallen asleep at the wheel

**Comment by jugglinmike** [comment]
All set!

**Comment by jugglinmike** [comment]
:fire: :smiling_imp: :fire:

**Comment by leobalter** [comment]
ecma262 is close to reach 666, maybe we should propose a `goto` syntax that mixes well with `with`.

"use with";
"use tco";
"use ptc";
